src/panels/lovelace/components/hui-action-editor.ts

```diff
@@ -115,51 +240,142 @@ export class HuiActionEditor extends LitElement {
             ></paper-input>
           `
         : ""}
-      ${this.config?.action === "call-service"
+      ${this._config?.action === "call-service"
         ? html`
             <ha-service-picker
               .hass=${this.hass}
               .value=${this._service}
               .configValue=${"service"}
               @value-changed=${this._valueChanged}
             ></ha-service-picker>
-            <b>
-              ${this.hass!.localize(
-                "ui.panel.lovelace.editor.action-editor.editor_service_data"
-              )}
-            </b>
+            ${this._service
+              ? html`<div class="secondary">
+                    ${computeServiceDescription(
+                      this.hass,
+                      this._service.split(".", 2)[0],
+                      this._service.split(".", 2)[1]
+                    )}
+                  </div>
+                  <div class="service-data-editor">
+                    <ha-yaml-editor
+                      .defaultValue=${this._service_data}
+                      .configValue=${"service_data"}
+                      @value-changed=${this._valueChanged}
+                    ></ha-yaml-editor>
+                  </div>
+                  <table class="attributes">
+                    <tr>
+                      <th>
+                        Parameter - Translate
+                      </th>
+                      <th>
+                        Description - Translate
+                      </th>
+                      <th>
+                        Example - Translate
+                      </th>
+                    </tr>
+                    ${repeat(
+                      computeServiceAttributes(
+                        this.hass,
+                        this._service.split(".", 2)[0],
+                        this._service.split(".", 2)[1]
+                      ),
+                      (attribute) => attribute.key,
+                      (attribute) =>
+                        html`
+                          <tr>
+                            <td><pre>${attribute.key}</pre></td>
+                            <td>${attribute.description}</td>
+                            <td>${attribute.example}</td>
+                          </tr>
+                        `
+                    )}
+                  </table>`
```

Comment:
This should be a separate shared element. We now have the same code in 3 places.

Reply:
And then we can also add the fill example data button :smile:

Reply:
Mentioned it above, but problem is that configs are different for automation editor vs lovelace actions. Maybe could do some translations when the data is received by the caller? Not sure.

Reply:
Regardless, I think I'll tackle that in a follow-up PR